File: sol/1352/1352.cpp
struct ProductOfNumbers {
    vector<int> m_seq;
    ProductOfNumbers(): m_seq(1, 1) {}
    
    void add(int num) {
        if (num == 0){
            m_seq.clear();
            m_seq.push_back(1);
        } else {
            m_seq.push_back(m_seq.back() * num);
        }
    }
    
    int getProduct(int k) {
        int n = m_seq.size();
        if (k >= n){
            return 0;
        }
        return m_seq.back() / m_seq[n - k - 1];
    }
};

File: sol/1352/1352.py
class ProductOfNumbers:
    def __init__(self):
        self.seq: list[int] = [1]

    def add(self, num: int) -> None:
        if num == 0:
            self.seq = [1]
        else:
            self.seq.append(self.seq[-1] * num)

    def getProduct(self, k: int) -> int:
        n = len(self.seq)
        if k >= n:
            return 0
        return self.seq[-1] // self.seq[n - k - 1]

